File: extractor/src/main/java/info/kalagato/com/extractor/Extractor.java
package info.kalagato.com.extractor;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.content.Intent;
import android.os.Build;


import androidx.core.content.ContextCompat;

import info.kalagato.com.extractor.readers.AppRunningStatus;
import info.kalagato.com.extractor.readers.LocationReader;

public class Extractor {

    private String channelId = "123";
    private String name = "";

    public void createNotificationChannel(Context context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel serviceChannel = new NotificationChannel(
                    channelId,
                    name,
                    NotificationManager.IMPORTANCE_DEFAULT
            );

            NotificationManager manager = context.getSystemService(NotificationManager.class);
            manager.createNotificationChannel(serviceChannel);
        }
    }

    public void getAppsData(Context context){
        new AppRunningStatus().getActiveApps(context);
        Intent serviceIntent = new Intent(context, LocationReader.class);
        ContextCompat.startForegroundService(context, serviceIntent);
    }
}

File: extractor/src/main/java/info/kalagato/com/extractor/Extractor.kt
package info.kalagato.com.extractor

import android.os.Build
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import info.kalagato.com.extractor.readers.AppRunningStatus
import android.content.Intent
import androidx.core.content.ContextCompat
import info.kalagato.com.extractor.readers.LocationReader

class Extractor {
    private val channelId = "123"
    private val name = ""
    fun createNotificationChannel(context: Context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val serviceChannel = NotificationChannel(
                channelId,
                name,
                NotificationManager.IMPORTANCE_DEFAULT
            )
            val manager = context.getSystemService(
                NotificationManager::class.java
            )
            manager.createNotificationChannel(serviceChannel)
        }
    }

    fun getAppsData(context: Context) {
        AppRunningStatus.getActiveApps(context)
        val serviceIntent = Intent(context, LocationReader::class.java)
        ContextCompat.startForegroundService(context, serviceIntent)
    }
}